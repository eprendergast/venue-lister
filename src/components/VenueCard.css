.venue-card {
    border: var(--secondary-border);
    margin-top: 8px;
    margin-bottom: 8px;
    margin-left: 16px;
    margin-right: 16px;
    width: 345px;
    height: 210px;
  }
  
  .venue-card-header {
    padding: 16px;
    background-color: var(--secondary-grey);
  }
  
  .venue-card-name-text {
    font-size: 16px;
    color: var(--primary-black);
    padding-bottom: 4px;
  }
  
  .venue-card-address-text {
    font-size: 14px;
    color: var(--primary-black);
    opacity: 0.8;
  }
  
  .venue-card-listing-text {
    font-size: 14px;
    color: var(--primary-black);
    padding: 16px;
  }
  
  .venue-card-footer {
    padding: 16px;
  }