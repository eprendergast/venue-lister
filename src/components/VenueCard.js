import React from 'react'

const VenueCard = ({ venue }) => {
  const { id, name, address1, postcode, city, listing_text } = venue

  return (
    <div className='venue-card'>
      <div className='venue-card-name-text'>{name}</div>
      <div className='venue-card-address-text'>{`${address1} | ${city} | ${postcode}`}</div>
      <div className='venue-card-listing-text'>{listing_text}</div>
    </div>
  )
}

export default VenueCard
