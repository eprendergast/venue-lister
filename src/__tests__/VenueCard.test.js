import React from 'react'
import renderer from 'react-test-renderer'
import VenueCard from '../components/VenueCard.js'

it('renders correctly', () => {
  const venue = {
    id: '835e83df-56c3-4808-8299-580bb4618ea8',
    name: 'Novotel Suites Den Haag City Center',
    address1: 'Grote Marktstraat 46',
    postcode: '2511',
    city: 'The Hague',
    listing_text:
      'Novotel Suites Den Haag City Center is located in the Nieuwe Haagsche Passage shopping arcade in the center of The Hague. The hotel has 118 suites for the guests, all are well equipped with all modern amenities.'
  }

  const tree = renderer.create(<VenueCard venue={venue} />).toJSON()
  expect(tree).toMatchSnapshot()
})
