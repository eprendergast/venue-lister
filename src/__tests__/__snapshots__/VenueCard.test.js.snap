// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="venue-card"
>
  <div
    className="venue-card-header"
  >
    <div
      className="venue-card-name-text"
    >
      Novotel Suites Den Haag City Center
    </div>
    <div
      className="venue-card-address-text"
    >
      Grote Marktstraat 46 | The Hague | 2511
    </div>
  </div>
  <div
    className="venue-card-listing-text"
  >
    Novotel Suites Den Haag City Center is located in the Nieuwe Haagsche Passage shopping arcade in the center of The Hague. The hotel has 118 suites for the guests, all are well equipped with all modern amenities.
  </div>
</div>
`;
